src/components/experimental/IndexProvider/utilities/hooks.ts

```diff
@@ -0,0 +1,31 @@
+import {useContext} from 'react';
+
+import {
+  IndexContext,
+  IndexRowContext,
+  IndexSelectionChangeContext,
+} from './context';
+
+export function useIndexRow() {
```

Comment:
what do you think of being a little more explicit with the naming and do `useIndexRowContext`, `useIndexContext`, ...